SWATCH-4001: Add swatch-utilization Quarkus service

Sgitario opened the pull request:
Jira issue: SWATCH-4001

## Description
Added the new swatch-utilization service with basic functionality:
- splunk
- kafka with a basic version of the utilization consumer
- health checks
- metrics
- component tests

## Testing

### Setup
1. podman compose up -d kafka kafka-bridge kafka-setup

### Steps
1. ./mvnw clean install -DskipTests
2. make swatch-utilization

### Verification
1. curl localhost:9018/health should be up and running

## Component tests
1. ./mvnw clean install -DskipTests
2. ./mvnw clean install -Pcomponent-tests -Dservice=swatch-utilization

vbusch commented on swatch-utilization/ct/README.md:
````diff
@@ -0,0 +1,41 @@
+Deploy only the necessary dependencies for a specific service. For example, for `swatch-utilization` (which only requires wiremock and kafka bridge, as Kafka comes by default):
+
+```bash
+bonfire deploy rhsm --source=appsre --ref-env insights-stage --component swatch-utilization --remove-dependencies swatch-utilization --component swatch-kafka-bridge
````
The bonfire deploy is failing with:
`deploy failed: component given for --remove-dependencies not found in app config: swatch-utilization`

vbusch reviewed (changes requested):
The ct/README.md is comprehensive.  I short README,nd should be included for the service itself.

Other areas look complete and well done.

I added a note to future SWATCH-4005 to handle the TODOs.

Sgitario commented:
@vbusch PR updated with:
- added a new component test to verify the metrics (plus, ability to check the metrics using the component test framework)
- configured the format command in makefile to also format the component tests sources
- added all the changes from your review

InsightsDroid commented:
### IQE Tests Summary Report
 3 failed, 309 passed, 2 skipped, 184 deselected, 70 warnings in 2209.55s (0:36:49) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions
|Failures|
|:-| 
|component/swatch_metrics_hbi/test_hbi_created_updated_events.py::test_cloud_provider_events[created-INSTANCE_CREATED]| 
|component/swatch_metrics_hbi/test_hbi_created_updated_events.py::test_cloud_provider_events[updated-INSTANCE_UPDATED]| 
|integration/swatch_tally/test_tally_openshift.py::test_validate_tally_on_openshift_cores_for_new_candlepin_facts[2-None]|

vbusch reviewed (approved):
LGTM
